blaze/compute/pandas.py

```diff
@@ -192,9 +192,18 @@ def compute_up(t, s, **kwargs):
     return result
 
 
-@dispatch(Distinct, (DataFrame, Series))
+@dispatch(Distinct, DataFrame)
 def compute_up(t, df, **kwargs):
-    return df.drop_duplicates().reset_index(drop=True)
+    return df.drop_duplicates(
+        subset=t.on if t.on else None,
+    ).reset_index(drop=True)
+
+
+@dispatch(Distinct, Series)
+def compute_up(t, s, **kwargs):
+    if t.on:
+        raise ValueError('series distinct cannot specify what to distinct on')
```

Comment:
the ndarray case makes sense for struct and recarrays but not for anything else so sharing the language doesn't make sense in every case.

i'd do something like this:

```python
if getattr(data.dtype, 'names', None) is None:
    raise ValueError('operation does not make sense')
else:
    raise NotImplementedError('makes sense, but numpy does not have an API for this operation')
```